src/ape_ethereum/ecosystem.py

```diff
@@ -337,8 +338,39 @@ def encode_calldata(self, abi: Union[ConstructorABI, MethodABI], *args) -> HexBy
         if not abi.inputs:
             return HexBytes("")
 
+        # Convert dicts and other objects to struct inputs
+        arguments: List = list(args)
+        for idx, ipt in enumerate(abi.inputs):
+            arg = arguments[idx]
+            if (
+                ipt.type == "tuple"
+                and ipt.components
+                and all(m.name for m in ipt.components)
+                and not isinstance(arg, tuple)
+            ):
+                if isinstance(arg, dict):
+                    arguments[idx] = tuple([arg[m.name] for m in ipt.components])
+
+                else:
+                    arg = [getattr(arg, m.name) for m in ipt.components if m.name]
+                    arguments[idx] = tuple(arg)
+
         input_types = [i.canonical_type for i in abi.inputs]
-        encoded_calldata = encode(input_types, args)
+
+        def convert(ls):
```

Comment:
idk, is there a better way to do this @fubuloubu ?
I tried using the tuple converter but it attempts to convert too much.